sky/skylet/constants.py

```diff
@@ -39,5 +39,5 @@
 # lifetime of the job.
 TASK_ID_LIST_ENV_VAR = 'SKYPILOT_TASK_IDS'
 
-SKYLET_VERSION = '1'
+SKYLET_VERSION = '2'
```

Comment:
TODO for self: read this

Reply:
Just updated the comment above, if you were curious about why we want to increase the version number.

Reply:
Don't see the comment yet?

I just read the skylet update code. Since we potentially kill a running skylet, will it cause any problems due to 
- (outside of this PR) ongoing events (autostop, etc.)?
- (this PR) scheduling and handling of spot jobs (JobSchedulerEvent, SpotJobUpdateEvent)?

Maybe another way to do it in the future is that we send an update signal to a running skylet, and it will gracefully exit.

Reply:
Oh, the comment was here https://github.com/skypilot-org/skypilot/pull/1982#issuecomment-1586383069

It should be ok for now to kill the skylet:
- For the autostop event, since the update will be called after the `sky launch`, the autostop should not be triggered. Even though the autostop is triggered during the `sky launch`, it is an undefined behavior, and killing the skylet and updating will not worsen it.
- For those events update, it will not be a problem at the moment, as those events only write to the database at most once for each task. Interrupting the event should not affect its correctness.

Despite what was said, it is a good idea to have the killing graceful, added a comment in the code.

Reply:
SGTM. Let's SHIP!